src/tx/staking/verifyStake.ts

```diff
@@ -173,5 +172,40 @@ export function verifyUnstakeTx(
     reason = `This nominee node is not found!`
   }
 
+  if (!isStakeUnlocked(nominatorAccount, nomineeAccount, shardus)) {
+    success = false
+    reason = `The stake is not unlocked yet!`
+  }
+
   return { success, reason }
 }
+
+function isStakeUnlocked(
+  nominatorAccount: WrappedEVMAccount,
+  nomineeAccount: NodeAccount2,
+  shardus: Shardus
+): boolean {
+  const stakeLockTime = AccountsStorage.cachedNetworkAccount.current.stakeLockTime
```

Comment:
we shuold not use cachedNetworkAccount here anymore. instead we should use networkAccount from wrappedState as this function gets called from apply.  reference: [](https://linear.app/shm/issue/SHARD-41/todo-review-safety-of-using-cached-network-account-in-apply-functions)